rllib/evaluation/sampler.py

```diff
@@ -1250,6 +1250,20 @@ def _process_policy_eval_results(
                 episode._set_last_action(agent_id, action)
 
             assert agent_id not in actions_to_send[env_id]
+            # Flag actions as immutable to notify the user when trying to change it
+            # and to avoid hardly traceable errors.
+            def make_action_immutable(obj):
```

Comment:
this looks awesome. what do you think we move this into rllib/utils/numpy.py?
I will try to reuse it for some connectors.

Reply:
> this looks awesome. what do you think we move this into rllib/utils/numpy.py? I will try to reuse it for some connectors.

@gjoliver Thanks! Yes I can do this. Regarding the connectors the @dataclass decorator could also become interesting as it makes classes immutable by the `__hash__()` function - I was thinking about using it here, but it was an overload